litmus-portal/backend/tools/self-deployer/deployer.go

```diff
@@ -75,44 +70,55 @@ func RegSelfCluster(server, pid string) (string, error) {
 }
 
 //Deploy function deploys the Manifest received by the RegSelfCluster function
-func Deploy(file string, kubeconfig *string) error {
-	resp, err := http.Get(file)
-	if err != nil {
-		log.Print(err.Error())
-		return err
-	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		log.Print(err.Error())
-		return err
-	}
-	log.Print(string(body))
-	response, err := k8s.CreateDeployment(body, kubeconfig)
-	if err != nil {
-		log.Println(err.Error())
+func Deploy(file string) error {
+	cmd := exec.Command("kubectl", "apply", "-f", file)
+	if output, err := cmd.Output(); err != nil {
+		log.Print("XX FAILED TO DEPLOY SUBSCRIBER COMPONENTS")
+		log.Print("ERROR : ", err.Error())
 		return err
+	} else {
+		log.Printf("\n%s", output)
+		log.Print("SUCCESSFULLY DEPLOYED SUBSCRIBER COMPONENTS")
 	}
-	responseData, err := json.Marshal(response)
-	if err != nil {
-		log.Println("err:", err)
-	}
-	log.Print("SUBSCRIBER DEPLOYMENT RESPONSE : ", string(responseData))
 	return nil
 }
 
 //Cleanup is responsible for deleting the self-deployer deployment after it has finished execution
-func CleanUp(ns, deplotment string) error {
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		log.Print(err.Error())
-		return err
-	}
-	client, err := kubernetes.NewForConfig(config)
-	err = client.AppsV1().Deployments(ns).Delete(context.TODO(), deplotment, metav1.DeleteOptions{})
-	if err != nil {
+func CleanUp(ns, deployment string) error {
+	cmd := exec.Command("kubectl", "delete", "deployment", deployment, "-n", ns)
+	if output, err := cmd.Output(); err != nil {
+		log.Print("XX FAILED TO CLEANUP SELF-DEPLOYER")
 		log.Print("ERROR : ", err.Error())
 		return err
+	} else {
+		log.Printf("\n%s", output)
+		log.Print("SUCCESSFUL CLEANUP OF SELF-DEPLOYER")
 	}
 	return nil
 }
+
+func main() {
+	active := os.Getenv("SELF_CLUSTER")
+	if strings.ToLower(active) == "true" {
+		server := os.Getenv("SERVER")
+		log.Print("Starting Self Deployer")
+		WaitForServer(server + "/query")
+		log.Print("Cluster Registration Initiated")
+		tk, err := RegSelfCluster(server+"/query", "00000")
+		if err != nil {
+			log.Panic(err.Error())
+		}
+		time.Sleep(10 * time.Second)
```

Comment:
Don't put specific sleep. You can put an if condition to check the tk

Reply:
We need to sleep for sometime because the clusterReg op may take some time to write to the disk, in normal human interaction a user would take some time after registering to actually go and apply the yaml, but in this case it is instantaneous. We can reduce the sleep time to 5s maybe but I kept 10s to be on the safe side.